src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -807,6 +808,12 @@ private void Parse()
             ExtractScheme(match);
             ExtractHosts(match);
 
+            if (_connect.HasValue && _directConnection.HasValue)
+            {
+                throw new MongoConfigurationException("Specifying both connect and directConnection is invalid.");
```

Comment:
Suggested error message change:

    throw new MongoConfigurationException("Connect and directConnection cannot both be specified.");